lib/flume/queue/api.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Flume.Queue.API do
```

Comment:
@vasuadari Can we keep this module abstract and just use it as a behaviour?

And move the implementation to a separate module ```Flume.Queue.DefaultAPI``` or something else?

Reply:
Sure